src/modules/eliteboard/types/EliteBoard.h

```diff
@@ -1,121 +1,95 @@
-  static const int scmPortDAdpNum = 3;
-  static const int scmPortEAdpNum = 4;
-  static const int scmPortFAdpNum = 5;
-  static const int scmPortGAdpNum = 6;
-  static const int scmPortHAdpNum = 7;
-  static const int scmPortIAdpNum = 8;
-  static const int scmPortJAdpNum = 9;
-  static const int scmPortKAdpNum = 10;
-
-  static const SAdapterInstanceDef scmAdapterInstances[];
-
-  static const SFBInterfaceSpec scmFBInterfaceSpec;
-
-  FORTE_PortAdapter &st_PortA() {
-    return (*static_cast<FORTE_PortAdapter *>(mAdapters[0]));
-  };
-
-  FORTE_PortAdapter &st_PortB() {
-    return (*static_cast<FORTE_PortAdapter *>(mAdapters[1]));
-  };
-
-  FORTE_PortAdapter &st_PortC() {
-    return (*static_cast<FORTE_PortAdapter *>(mAdapters[2]));
-  };
+/*************************************************************************
+ *** FORTE Library Element
+ ***
+ *** This file was generated using the 4DIAC FORTE Export Filter V1.0.x NG!
+ ***
+ *** Name: EliteBoard
+ *** Description: Service Interface Function Block Type
+ *** Version:
+ ***     1.0: 2021-02-22/Jonathan Lainer -  - Initial Contribution
+ *************************************************************************/
+
+#pragma once
 
-  FORTE_PortAdapter &st_PortD() {
-    return (*static_cast<FORTE_PortAdapter *>(mAdapters[3]));
-  };
-
-  FORTE_PortAdapter &st_PortE() {
-    return (*static_cast<FORTE_PortAdapter *>(mAdapters[4]));
-  };
-
-  FORTE_PortAdapter &st_PortF() {
-    return (*static_cast<FORTE_PortAdapter *>(mAdapters[5]));
-  };
-
-  FORTE_PortAdapter &st_PortG() {
-    return (*static_cast<FORTE_PortAdapter *>(mAdapters[6]));
-  };
-
-  FORTE_PortAdapter &st_PortH() {
-    return (*static_cast<FORTE_PortAdapter *>(mAdapters[7]));
-  };
-
-  FORTE_PortAdapter &st_PortI() {
-    return (*static_cast<FORTE_PortAdapter *>(mAdapters[8]));
-  };
-
-  FORTE_PortAdapter &st_PortJ() {
-    return (*static_cast<FORTE_PortAdapter *>(mAdapters[9]));
-  };
-
-  FORTE_PortAdapter &st_PortK() {
-    return (*static_cast<FORTE_PortAdapter *>(mAdapters[10]));
-  };
-
-
-  void executeEvent(TEventID paEIID, CEventChainExecutionThread *const paECET) override;
- 
-  void setConfig() {}
-  
-  EliteBoardDeviceController& mEventHandler;
-  int mCurrentAdapterIndex = 0;
-  static const int mAdapterCount = scmPortKAdpNum + 1;
-
-  forte::core::io::IODeviceController *
-  createDeviceController(CDeviceExecution &paDeviceExecution);
-  FORTE_PortAdapter &getPortAdapterByIndex(int index);
-  bool configurePortFB(int index);
+#include "funcbloc.h"
+#include "PortAdapter_adp.h"
+#include "forte_dword.h"
+#include "iec61131_functions.h"
+#include "forte_array_common.h"
+#include "forte_array.h"
+#include "forte_array_fixed.h"
+#include "forte_array_variable.h"
 
-public:
+#include "../handler/EliteBoardDeviceController.h"
 
-  FORTE_EliteBoard(const CStringDictionary::TStringId paInstanceNameId, 
-      forte::core::CFBContainer &paContainer);
+class FORTE_EliteBoard final : public CFunctionBlock {
```

Comment:
This should still be a forte::core::io::IOConfigFBController

Reply:
although we are currently not using any IOConfigFBController functionality. Maybe this is then something for later.